### PR TITLE
fix: nil pointer in GC

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlike to be merged.
* [x] Optional. I've added My organization is added to the README.
* [ ] I've signed the CLA and required builds are green. 

This PR adds:

1. Fixes the nil deref in the GC.
1. Adds a new `no-db` test suite and set-up options, e.g. `make start DB=no-db`

### CODE REVIEW

File: workflow/controller/config_controller.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func (wfc *WorkflowController) updateConfig(cm *apiv1.ConfigMap) error {
 		}
 	}
 	wfc.session = nil
-	wfc.offloadNodeStatusRepo = nil
+	wfc.offloadNodeStatusRepo = sqldb.ExplosiveOffloadNodeStatusRepo
 	wfc.wfArchive = sqldb.NullWorkflowArchive
 	persistence := wfc.Config.Persistence
```
Comment (sarabala1979) [inline workflow/controller/config_controller.go]: We need to do nil check here

Comment (sarabala1979) [inline workflow/controller/config_controller.go]: We need to do nil check here

Comment (alexec) [reply workflow/controller/config_controller.go]: no - at this line of code `wc.Config` cannot be nil

Comment (sarabala1979) [reply workflow/controller/config_controller.go]: ok

**Pull request merged**